packet/cloud.go

```diff
@@ -129,6 +135,11 @@ func (c *cloud) Initialize(clientBuilder cloudprovider.ControllerClientBuilder,
 			serviceReconcilers = append(serviceReconcilers, s)
 		}
 	}
+
+	if c.controlPlaneEndpointManager != nil {
+		nodeReconcilers = append(nodeReconcilers, c.controlPlaneEndpointManager.Reconciler())
+	}
+
```

Comment:
Then this whole thing can go away.

Reply:
And you can add `controlPlaneEndpointManager` to `services` slice, and it is all part of the same pattern.